entrypoint.py

```diff
@@ -0,0 +1,267 @@
+#! /usr/bin/env python3
+
+from jinja2 import Environment, FileSystemLoader
+from argparse import ArgumentParser
+from contextlib import ExitStack
+import requests
+import zipfile
+import shutil
+import json
+import csv
+import os
+
+DIGIKEY_API_URL_BASE = "https://api.digikey.com"
+DIGIKEY_API_AUTH_ENDPOINT = DIGIKEY_API_URL_BASE + "/v1/oauth2/token"
+DIGIKEY_API_V4_KEYWORD_SEARCH_ENDPOINT = (
+    DIGIKEY_API_URL_BASE + "/products/v4/search/keyword"
+)
+
+
+################################################################################
+class ComponentData:
+    def __init__(self):
+        self.associated_refdes = ""
+        self.part_description = None
+        self.mfr_name = None
+        self.mfr_part_number = None
+        self.photo_url = None
+        self.datasheet_url = None
+        self.product_url = None
+        self.qty_available = None
+        self.lifecycle_status = None
+        self.eol_status = None
+        self.discontinued_status = None
+        self.pricing = None
+        self.package_case = None
+        self.supplier_device_package = None
+        self.operating_temp = None
+        self.xy_size = None
+        self.height = None
+        self.thickness = None
+
+
+################################################################################
+def get_access_token(url, client_id, client_secret):
+    # Populate request header
+    headers = {
+        "Content-Type": "application/x-www-form-urlencoded",
+    }
+    # Post the request and get the response
+    response = requests.post(
+        url,
+        data={"grant_type": "client_credentials"},
+        headers=headers,
+        auth=(client_id, client_secret),
+    )
+    # Populate the access token return value
+    access_token = (
+        response.json()["access_token"] if response.status_code == 200 else None
+    )
+    # Return response status code and access token
+    return (response.status_code, access_token)
```

Comment:
I don't see this response code being used anywhere. Is there a reason we're returning it?

Reply:
Good catch, the response code is being used in the API query endpoint but not in the auth. I'll add the check to auth to exit if auth fails.